src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/Null.ReadTests.cs

```diff
@@ -222,18 +222,39 @@ public static void DeserializeDictionaryWithNullValues()
             }
         }
 
-        [Fact]
-        public static void InvalidRootOnRead()
+        [Theory]
+        [InlineData(typeof(UIntPtr))]
+        [InlineData(typeof(IntPtr))]
+        [InlineData(typeof(Type))]
+        [InlineData(typeof(int[,]))]
+        [InlineData(typeof(System.Delegate))]
+        public static void InvalidRootOnRead(Type type)
         {
-            Assert.Throws<NotSupportedException>(() => JsonSerializer.Deserialize<int[,]>("null"));
+            if (type.IsValueType)
+            {
+                // Unsupported struct types throw
+                Assert.Throws<NotSupportedException>(() => JsonSerializer.Deserialize("null", type));
```

Comment:
can we add same test with custom converter added through options?

Reply:
I don't think custom converters are in scope for this particular test suite or this change. We have a separate test suite tracking custom converters and how they handle null values.